hs_core/views/discovery_json_view.py

```diff
@@ -0,0 +1,41 @@
+import simplejson as json
+from django.http import HttpResponse
+from haystack.query import SearchQuerySet
+from django import forms
+from haystack.forms import FacetedSearchForm
+from haystack.generic_views import FacetedSearchView
+from django.core import serializers
+from hs_core.discovery_form import DiscoveryForm
+
+class DiscoveryJsonView(FacetedSearchView):
+    facet_fields = ['author', 'subjects', 'resource_type', 'public', 'owners_names', 'discoverable']
+    form_class = DiscoveryForm
+
+    def form_valid(self, form):
+        coor_values = []
+        coordinate_dictionary = []
+        self.queryset = form.search()
+        if len(self.request.GET):
+            for result in self.get_queryset():
+                json_obj = {}
+                json_obj['title'] = result.object.title
+                json_obj['get_absolute_url'] = result.object.get_absolute_url()
+                for coverage in result.object.metadata.coverages.all():
```

Comment:
From my point of view, I think if we want to pull data from Solr, we need to do that via Haystack. However, the default data type returned from Haystack is in HTML template format. We want to have JSON data format for building our map. That's why we need to have this view. I think Solr is just a search engine to index the data. We need to pull data via Haystack. I am not sure whether we could pull JSON format data directly from Solr. Maybe Alva could offer some suggestion here. @alvacouch